oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -121,6 +121,49 @@ pub enum ReadStatus {
     NeedsCapacity(usize, usize),
 }
 
+/// Information for managing an associated server.
+pub struct AuxServer {
+    pub name: String,
+    pub join_handle: Option<JoinHandle<()>>,
+    pub notify_sender: Option<tokio::sync::oneshot::Sender<()>>,
+}
+
+impl AuxServer {
+    fn new<F: FnOnce(u16, Arc<Runtime>, tokio::sync::oneshot::Receiver<()>) + 'static + Send>(
+        name: &str,
+        port: u16,
+        runtime: Arc<Runtime>,
+        f: F,
+    ) -> Self {
+        let (notify_sender, notify_receiver) = tokio::sync::oneshot::channel::<()>();
+        info!("spawning {} server on new thread", name);
+        let join_handle = thread::Builder::new()
+            .name(format!("{}-server", name))
+            .spawn(move || f(port, runtime, notify_receiver))
+            .expect("failed to spawn introspection thread");
+        AuxServer {
+            name: name.to_string(),
+            join_handle: Some(join_handle),
+            notify_sender: Some(notify_sender),
+        }
+    }
+}
+
+impl Drop for AuxServer {
+    fn drop(&mut self) {
+        let join_handle = std::mem::replace(&mut self.join_handle, None);
```

Comment:
Does `self.join_handle.take()` work here?

Reply:
Yes! I knew there was a better way of doing this, but it escaped my memory (and my Google search)